### PR TITLE
Use special report suffix for internal ci and add script for cloud-to-prod

### PR DESCRIPTION
**Pull request by adelez**

### CODE REVIEW

**Comment by jtattermusch** [comment]
Please double check that Jenkins jobs for interop tests (PR and master) still display the reports with this change. You might need to change the pattern for "JUnit plugin".
E.g. https://grpc-testing.appspot.com/job/gRPC_interop_pull_requests/configure uses pattern "**/report.xml" which needs to be updated (see Basic tests jobs for reference).

File: tools/internal_ci/linux/grpc_interop_toprod.cfg [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+# Config file for the internal CI (in protobuf text format)
+
+# Location of the continuous shell script in repository.
+build_file: "grpc/tools/internal_ci/linux/grpc_interop_toprod.sh"
+# grpc_interop tests can take 6+ hours to complete.
+timeout_mins: 480
+action {
+  define_artifacts {
+    regex: "**/report.xml"
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_toprod.cfg]: regex: "**/*sponge_log.xml"

https://github.com/grpc/grpc/blob/72cdf6f08242251810edae2df0b3f261afc73c2e/tools/internal_ci/macos/grpc_master.cfg#L22

File: tools/internal_ci/linux/grpc_interop_toprod.cfg [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Config file for the internal CI (in protobuf text format)
+
+# Location of the continuous shell script in repository.
+build_file: "grpc/tools/internal_ci/linux/grpc_interop_toprod.sh"
+# grpc_interop tests can take 6+ hours to complete.
+timeout_mins: 480
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_toprod.cfg]: nit: timeout should be way shorter.

File: tools/internal_ci/linux/grpc_interop_toprod.cfg (hunk 1 above)
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_toprod.cfg]: same for _tocloud jobs

Comment (adelez) [inline tools/internal_ci/linux/grpc_interop_toprod.cfg]: Done. I'm just naming the interop result to sponge_log.xml.

**Comment by adelez** [comment]
Base Tests Mac failures: #11223
Basic Tests Windows (internal CI): #11482

**Comment by adelez** [comment]
Bazel Full Build & Tests failures: ignore for now.

File: tools/internal_ci/linux/grpc_interop_toprod.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# Enter the gRPC repo root
+cd $(dirname $0)/../../..
+
+source tools/internal_ci/helper_scripts/prepare_build_linux_rc
+source tools/internal_ci/helper_scripts/prepare_build_interop_rc
+
+tools/run_tests/run_interop_tests.py \
+    -l all \
+    --cloud_to_prod \
+    --cloud_to_prod_auth \
```
Comment (jtattermusch) [inline tools/internal_ci/linux/grpc_interop_toprod.sh]: Btw, one of the --cloud_to_prod_auth  tests is a "service account key" credentials test and that one won't pass until we are able to provide the service account key.
One of the ways we came up with was:
- store the secret .json in GCS, allowing 
- download the secret with gsutil, using the GCE compute engine credentials to authenticate with GCS (and gaining access to the secret).
- I'll send more info offline

**Pull request merged**